Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp

```diff
@@ -321,12 +320,15 @@ void ThreadedCompositor::renderLayerTree()
 
     uint32_t compositionRequestID = m_compositionRequestID.load();
     m_compositionResponseID = compositionRequestID;
-    if (!m_didRenderFrameTimer.isActive())
-        m_didRenderFrameTimer.startOneShot(0_s);
 #if !HAVE(OS_SIGNPOST) && !USE(SYSPROF_CAPTURE)
     UNUSED_VARIABLE(compositionRequestID);
 #endif
 
+    ASSERT(!m_didCompositeRunLoopObserver->isScheduled());
+    RunLoop::mainSingleton().dispatch([this, protectedThis = Ref { *this }] {
+        m_didCompositeRunLoopObserver->schedule();
+    });
```

Comment:
schedule is not thread safe? if we have to dispatch a source in the main thread, then the timer was better since it has higher priority. The idea of the observer here is precisely avoid scheduling a source in the main thread run loop.

Reply:
I think we can update RunLoopObserver::schedule to handle the platform run loop. Then, we can schedule from the compositor thread but passing the main run loop.

Reply:
Needed to adjust things a bit for cross-thread scheduling, please have a look :-)